test_adjoint/test_swe_adjoint.py

```diff
@@ -93,6 +93,8 @@ def setup_unsteady():
     # automatically switches off matnest when using 'lu',
     # firedrake-adjoint does not remember that in the adjoint solve
     parameters['matnest'] = False
+    # coffee optimisation currently broken for this case:
+    parameters['coffee'] = {}
```

Comment:
No, neither did I. They're bugs really. One is fixed already by Fabio:
https://github.com/coneoproject/COFFEE/issues/81
and the other is a missing feature in firedrake-adjoint:
https://bitbucket.org/dolfin-adjoint/dolfin-adjoint/issues/43/firedrake-adjoint-does-not-respect-nest